pkg/resource/file_system/hooks.go

```diff
@@ -375,4 +366,218 @@ func customPreCompare(
+
+	// Check if filesystem is active before performing replication operations
+	if !filesystemActive(latest) {
+		return requeueWaitState(latest)
+	}
+
+	if !replicationConfigurationActive(latest) {
+		return requeueWaitReplicationConfiguration
+	}
+
+	desiredHasReplication := len(desired.ko.Spec.ReplicationConfiguration) == 1
+	latestHasReplication := len(latest.ko.Spec.ReplicationConfiguration) == 1
+
+	if len(desired.ko.Spec.ReplicationConfiguration) > 1 {
+		rlog.Info("Invalid replication configuration", 
+			"replicationCount", len(desired.ko.Spec.ReplicationConfiguration),
+			"replicationDestinations", desired.ko.Spec.ReplicationConfiguration)
+		msg := fmt.Errorf("EFS replication configuration only allows one replica, got %d replicas", 
+			len(desired.ko.Spec.ReplicationConfiguration))
+		return ackerr.NewTerminalError(msg)
+	}
+
+	if desiredHasReplication && latestHasReplication {
+		// Delete existing then create new (AWS doesn't support direct update)
+		err = rm.deleteReplicationConfiguration(ctx, latest)
+		if err != nil {
+			return err
+		}
+		err = rm.createReplicationConfiguration(ctx, desired)
+		if err != nil {
+			return err
+		}
+	}
+
+	// If desired has replication but latest doesn't, create it
+	if desiredHasReplication && !latestHasReplication {
+		err = rm.createReplicationConfiguration(ctx, desired)
+		if err != nil {
+			return err
+		}
+	}
+
+	// If desired doesn't have replication but latest does, delete it
+	if !desiredHasReplication && latestHasReplication {
+		err = rm.deleteReplicationConfiguration(ctx, desired)
+		if err != nil {
+			return err
+		}
+	}
+
+	return requeueWaitReplicationConfiguration
+}
+
+// createReplicationConfiguration creates replication configuration for the file system
+func (rm *resourceManager) createReplicationConfiguration(ctx context.Context, r *resource) (err error) {
+	rlog := ackrtlog.FromContext(ctx)
+	exit := rlog.Trace("rm.createReplicationConfiguration")
+	defer func() { exit(err) }()
+
+	// Convert DestinationToCreate to Smithy SDK format
+	destinations := make([]svcsdktypes.DestinationToCreate, 0, len(r.ko.Spec.ReplicationConfiguration))
+	for _, dest := range r.ko.Spec.ReplicationConfiguration {
+		sdkDest := svcsdktypes.DestinationToCreate{}
+		if dest.Region != nil {
+			sdkDest.Region = dest.Region
+		}
+		if dest.AvailabilityZoneName != nil {
+			sdkDest.AvailabilityZoneName = dest.AvailabilityZoneName
+		}
+		if dest.KMSKeyID != nil {
+			sdkDest.KmsKeyId = dest.KMSKeyID
+		}
+		if dest.FileSystemID != nil {
+			sdkDest.FileSystemId = dest.FileSystemID
+		}
+		if dest.RoleARN != nil {
+			sdkDest.RoleArn = dest.RoleARN
+		}
+		destinations = append(destinations, sdkDest)
+	}
+
+	_, err = rm.sdkapi.CreateReplicationConfiguration(
+		ctx,
+		&svcsdk.CreateReplicationConfigurationInput{
+			SourceFileSystemId: r.ko.Status.FileSystemID,
+			Destinations:       destinations,
+		},
+	)
+	rm.metrics.RecordAPICall("CREATE", "CreateReplicationConfiguration", err)
```

Comment:
This is part of the update workflow for EFS Filesystem
```suggestion
	rm.metrics.RecordAPICall("UPDATE", "CreateReplicationConfiguration", err)
```